src/NATS.Client.Core/Internal/SslStreamConnection.cs

```diff
@@ -115,9 +115,14 @@ public void SignalDisconnected(Exception exception)
     public async Task AuthenticateAsClientAsync(NatsUri uri, TimeSpan timeout)
     {
         var options = await _tlsOpts.AuthenticateAsClientOptionsAsync(uri).ConfigureAwait(true);
+        var networkStream = new NetworkStream(_socket, true);
+
 #if NETSTANDARD2_0
+        if (_sslStream != null)
+            _sslStream?.Dispose();
```

Comment:
Should we handle any exceptions here?

Reply:
I'm not sure, but there shouldn't be any issues. The only issue is there shouldn't be elvis operator in line 122 as I've already checked for null on line 121